gdk/common/CaseInsensitive.py

```diff
@@ -0,0 +1,98 @@
+import json
+import yaml
+from pathlib import Path
+
+from requests.structures import CaseInsensitiveDict as _CaseInsensitiveDict
+
+
+class CaseInsensitiveDict(_CaseInsensitiveDict):
+    def __init__(self, data=None, **kwargs):
+        super().__init__(data, **kwargs)
+        _dict = _CaseInsensitiveDict(data)
+        self._convert_nested_dict(_dict)
+        self.update(_dict)
+
+    def to_dict(self):
+        _dict = dict(self)
+        self._convert_nested_case_insensitive_dict(_dict)
+        return _dict
+
+    def update_value(self, key, value):
+        if key.lower() in self._store:
+            key = self._store[key.lower()][0]
+        self._store[key.lower()] = (key, value)
+
+    def _convert_nested_dict(self, case_insensitive_dict: _CaseInsensitiveDict) -> None:
+        for key, value in case_insensitive_dict.items():
+            if isinstance(value, dict):
+                case_insensitive_dict.update({key: CaseInsensitiveDict(value)})
+            elif isinstance(value, list):
+                case_insensitive_dict.update({key: [CaseInsensitiveDict(val) for val in value if isinstance(val, dict)]})
+
+    def _convert_nested_case_insensitive_dict(self, dictObj: dict) -> dict:
+        for key, value in dictObj.items():
+            if isinstance(value, CaseInsensitiveDict):
+                dictObj.update({key: self._convert_nested_case_insensitive_dict(dict(value))})
+            elif isinstance(value, list):
+                dictObj.update(
+                    {
+                        key: [
+                            self._convert_nested_case_insensitive_dict(dict(val))
+                            for val in value
+                            if isinstance(val, CaseInsensitiveDict)
+                        ]
+                    }
+                )
+        return dictObj
+
+
+class CaseInsensitiveRecipeFile:
+    def write(self, file_path: Path, content: CaseInsensitiveDict) -> None:
```

Comment:
I am ok with this given we only support 2 formats but if we ever get to more than 2 and support something like `toml` or `xml` this needs to be refactored.